apps/api/src/domain/models/shared/common-commands/delete-resource/delete-resource.command.integration.spec.ts

```diff
@@ -0,0 +1,46 @@
+import { CommandHandlerService } from '@coscrad/commands';
+import { INestApplication } from '@nestjs/common';
+import setUpIntegrationTest from '../../../../../app/controllers/__tests__/setUpIntegrationTest';
+import { IIdManager } from '../../../../../domain/interfaces/id-manager.interface';
+import generateDatabaseNameForTestSuite from '../../../../../persistence/repositories/__tests__/generateDatabaseNameForTestSuite';
+import TestRepositoryProvider from '../../../../../persistence/repositories/__tests__/TestRepositoryProvider';
+import { CommandAssertionDependencies } from '../../../__tests__/command-helpers/types/CommandAssertionDependencies';
+
+const commandType = 'DELETE_RESOURCE';
+
+describe(commandType, () => {
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let app: INestApplication;
+
+    let idManager: IIdManager;
+
+    let commandAssertionDependencies: CommandAssertionDependencies;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager, app } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+            }));
+
+        commandAssertionDependencies = {
+            testRepositoryProvider,
+            idManager,
+            commandHandlerService,
+        };
+    });
+
+    afterAll(async () => {
+        await app.close();
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
```

Comment:
- Happy Path (the resource exists and has not yet been deleted.
- Invalid Cases
    - The resource does not exist
    - The resource has already been deleted (similar to does not exist but with different setup)

Reply:
Just wondering about checking for existing connections to the resource